phoenix-core/src/main/java/org/apache/phoenix/cache/ServerCacheClient.java

```diff
@@ -394,4 +420,82 @@ public static String idToString(byte[] uuid) {
+                        @Override
+                        public AddServerCacheResponse call(ServerCachingService instance) throws IOException {
+                            ServerRpcController controller = new ServerRpcController();
+                            BlockingRpcCallback<AddServerCacheResponse> rpcCallback = new BlockingRpcCallback<AddServerCacheResponse>();
+                            AddServerCacheRequest.Builder builder = AddServerCacheRequest.newBuilder();
+                            final byte[] tenantIdBytes;
+                            if (cacheUsingTable.isMultiTenant()) {
+                                try {
+                                    tenantIdBytes = connection.getTenantId() == null ? null
+                                            : ScanUtil.getTenantIdBytes(cacheUsingTable.getRowKeySchema(),
+                                                    cacheUsingTable.getBucketNum() != null, connection.getTenantId(),
+                                                    cacheUsingTable.getViewIndexId() != null);
+                                } catch (SQLException e) {
+                                    throw new IOException(e);
+                                }
+                            } else {
+                                tenantIdBytes = connection.getTenantId() == null ? null
+                                        : connection.getTenantId().getBytes();
+                            }
+                            if (tenantIdBytes != null) {
+                                builder.setTenantId(ByteStringer.wrap(tenantIdBytes));
+                            }
+                            builder.setCacheId(ByteStringer.wrap(cacheId));
+                            builder.setCachePtr(org.apache.phoenix.protobuf.ProtobufUtil.toProto(cachePtr));
+                            builder.setHasProtoBufIndexMaintainer(true);
+                            ServerCacheFactoryProtos.ServerCacheFactory.Builder svrCacheFactoryBuider = ServerCacheFactoryProtos.ServerCacheFactory
+                                    .newBuilder();
+                            svrCacheFactoryBuider.setClassName(cacheFactory.getClass().getName());
+                            builder.setCacheFactory(svrCacheFactoryBuider.build());
+                            builder.setTxState(ByteStringer.wrap(txState));
+                            instance.addServerCache(controller, builder.build(), rpcCallback);
+                            if (controller.getFailedOn() != null) { throw controller.getFailedOn(); }
+                            return rpcCallback.get();
+                        }
+                    });
+        } catch (Throwable t) {
+            throw new Exception(t);
+        }
+        if (results != null && results.size() == 1) { return results.values().iterator().next().getReturn(); }
+        return false;
+
+    }
+    
+    public static ServerCache getCacheForId(List<ServerCache> caches, Long cacheId) {
```

Comment:
How about using a Map<ImmutableBytesPtr,ServerCache> instead of a List and switching id to ImmutableBytesPtr? Then you can get rid of this static method.